src/link/preview.js

```diff
@@ -0,0 +1,56 @@
+import { Popover } from '@wordpress/components';
+import { useState, useEffect } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { getSiteLanguage } from './utils';
+import { getPreviewHtml } from 'wikipedia-preview'
+
+export const PreviewEditUI = ( {
+	anchorRef,
+  title,
+	activePreview,
+	onClose,
+	onEdit,
+	onRemove
+} ) => {
+	const [previewHtml, setPreviewHtml] = useState(null);
+	const lang = getSiteLanguage()
+
+	useEffect( () => {
+		getPreviewHtml(title, lang).then(setPreviewHtml);
```

Comment:
if you are looking for the request with callback method, i [have here copied (the search pr)](https://github.com/wikimedia/wikipediapreview-wordpress/pull/32/files#diff-daebd7a1bbd53bdf6b0999736a2e74d0c608e3a916fe288daf96d636241ccf22R98) from kaios code